modules/examples/src/main/scala/dev/profunktor/redis4cats/RedisPoolDemo.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2018-2021 ProfunKtor
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dev.profunktor.redis4cats
+
+import cats.effect.IO
+import dev.profunktor.redis4cats.connection._
+import dev.profunktor.redis4cats.effect.Log.NoOp._
+import io.lettuce.core.RedisCommandExecutionException
+import org.typelevel.keypool.KeyPool
+
+object RedisPoolDemo extends LoggerIOApp {
+  import Demo._
+
+  val usernameKey = "test"
+  val numericKey  = "numeric"
+
+  val showResult: Option[String] => IO[Unit] =
+    _.fold(IO.println(s"Not found key: $usernameKey"))(IO.println)
+
+  // simple strings program
+  def p1(stringPool: KeyPool[IO, Unit, RedisCommands[IO, String, String]]): IO[Unit] = {
+    import dev.profunktor.redis4cats.Redis.Pool._
+    stringPool.withRedisCommands { redis =>
+      for {
+        x <- redis.get(usernameKey)
+        _ <- showResult(x)
+        _ <- redis.set(usernameKey, "some value")
+        y <- redis.get(usernameKey)
+        _ <- showResult(y)
+        _ <- redis.setNx(usernameKey, "should not happen")
+        w <- redis.get(usernameKey)
+        _ <- showResult(w)
+      } yield ()
+    }
+  }
+
+  // proof that you can still get it wrong with `incr` and `decr`, even if type-safe
+  def p2(
+      stringPool: KeyPool[IO, Unit, RedisCommands[IO, String, String]],
+      longPool: KeyPool[IO, Unit, RedisCommands[IO, String, Long]]
+  ): IO[Unit] = {
+    import dev.profunktor.redis4cats.Redis.Pool._
+    stringPool.withRedisCommands { redis =>
+      longPool.withRedisCommands { redisN =>
+        for {
+          x <- redis.get(numericKey)
+          _ <- showResult(x)
+          _ <- redis.set(numericKey, "not a number")
+          y <- redis.get(numericKey)
+          _ <- showResult(y)
+          _ <- redisN.incr(numericKey).attempt.flatMap {
+                case Left(e: RedisCommandExecutionException) =>
+                  IO(assert(e.getMessage == "ERR value is not an integer or out of range"))
+                case _ =>
+                  IO.raiseError(new Exception("Expected error"))
+              }
+          w <- redis.get(numericKey)
+          _ <- showResult(w)
+        } yield ()
+      }
+    }
+  }
+
+  val program: IO[Unit] = {
+    val res: fs2.Stream[IO, Unit] =
```

Comment:
`fs2.Stream` is the default in the codebase, please add the import at the top and use as `Stream` everywhere